functions/snippets/src/test/java/HelloGcsTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START functions_storage_unit_test]
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.google.common.testing.TestLogHandler;
+import com.google.gson.Gson;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.logging.Logger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+/**
+ * Unit tests for HelloGcs.
+ */
+@PrepareForTest(HelloGcs.class)
+public class HelloGcsTest {
+  @Mock private HttpRequest request;
+  @Mock private HttpResponse response;
+
+  // Use GSON (https://github.com/google/gson) to parse JSON content.
+  private Gson gson = new Gson();
+
+  private EnvironmentVariables environmentVariables;
+
+  private static final TestLogHandler logHandler = new TestLogHandler();
```

Comment:
Done throughout - marking these comments as resolved.

Reply:
Actually (while we are nitting), Google Style does not say that all `static final` fields should be spelled with `CONSTANT_CASE`. Only [truly immutable values](https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names) should be, which `TestLogHandler` and `Logger` are not.

Reply:
Ah interesting. I didn't know the Google Style Guide specified that - I've always just followed the [traditional Java conventions](https://stackoverflow.com/a/7259738/9414803) where anything `final` is considered a constant since the reference is immutable.